File: sdk/samples/WCFSamples/TechnologySamples/Scenario/WorkflowServices/Conversations/CS/Shipper1/ShipperWorkflow.cs
//  Copyright (c) Microsoft Corporation. All rights reserved.

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections.Generic;
using System.Drawing;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;
using System.Xml;

namespace Microsoft.WorkflowServices.Samples
{
	public sealed partial class ShipperWorkflow: SequentialWorkflowActivity
	{
		public ShipperWorkflow()
		{
			InitializeComponent();
		}

        public PurchaseOrder order = new PurchaseOrder();
        public IDictionary<string, string> supplierContext = default(IDictionary<string, string>);
        public string supplierAck = default(string);

        private void AcceptQuoteRequest(object sender, EventArgs e)
        {
            Console.ForegroundColor = ConsoleColor.White;
            Console.WriteLine("Received ShippingQuote Request");
            this.supplierAck = "Working on quote...";
            Console.ResetColor();
            this.SendShippingQuote.Context = this.supplierContext;
        }

        public ShippingQuote quote = new ShippingQuote();

        private void PrepareQuote(object sender, SendActivityEventArgs e)
        {
            Random rand = new Random();
            quote.ShippingCost = rand.Next(45, 90);
            quote.EstimatedShippingDate = DateTime.Now.AddDays(rand.Next(2,5));
            Console.ForegroundColor = ConsoleColor.White;
            Console.WriteLine("Sending ShippingQuote ");
            Console.WriteLine("Cost: $" + quote.ShippingCost);
            Console.WriteLine("ShipDate: " + quote.EstimatedShippingDate);
            Console.ForegroundColor = ConsoleColor.Gray;
            foreach (KeyValuePair<string, string> item in supplierContext)
                Console.WriteLine(item.Key + ": " + item.Value.Substring(0,13));
            Console.ResetColor();
        }

	}

}

File: sdk/samples/WCFSamples/TechnologySamples/Scenario/WorkflowServices/Conversations/VB/Shipper1/ShipperWorkflow.vb
' Copyright (c) Microsoft Corporation. All rights reserved.

Imports System
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Collections.Generic
Imports System.Drawing
Imports System.Workflow.ComponentModel.Compiler
Imports System.Workflow.ComponentModel.Serialization
Imports System.Workflow.ComponentModel
Imports System.Workflow.ComponentModel.Design
Imports System.Workflow.Runtime
Imports System.Workflow.Activities
Imports System.Workflow.Activities.Rules
Imports System.Xml

Namespace Microsoft.WorkflowServices.Samples
	Public NotInheritable Partial Class ShipperWorkflow
		Inherits SequentialWorkflowActivity
		Public Sub New()
			InitializeComponent()
		End Sub

		Public order As New PurchaseOrder()
        Public supplierContext As IDictionary(Of String, String) = Nothing
		Public supplierAck As String = Nothing

		Private Sub AcceptQuoteRequest(ByVal sender As Object, ByVal e As EventArgs)
			Console.ForegroundColor = ConsoleColor.White
			Console.WriteLine("Received ShippingQuote Request")
			Me.supplierAck = "Working on quote..."
			Console.ResetColor()
			Me.SendShippingQuote.Context = Me.supplierContext
		End Sub

		Public quote As New ShippingQuote()

		Private Sub PrepareQuote(ByVal sender As Object, ByVal e As SendActivityEventArgs)
			Dim rand As New Random()
			quote.ShippingCost = rand.[Next](45, 90)
			quote.EstimatedShippingDate = DateTime.Now.AddDays(rand.[Next](2, 5))
			Console.ForegroundColor = ConsoleColor.White
			Console.WriteLine("Sending ShippingQuote ")
            Console.WriteLine("Cost: $" + quote.ShippingCost.ToString())
            Console.WriteLine("ShipDate: " + quote.EstimatedShippingDate.ToString())
			Console.ForegroundColor = ConsoleColor.Gray
            For Each item As KeyValuePair(Of String, String) In supplierContext
                Console.WriteLine(item.Key & ": " & item.Value.Substring(0, 13))
            Next
			Console.ResetColor()
		End Sub

	End Class

End Namespace